ui/src/hooks/useProofRequest.ts

```diff
@@ -27,16 +29,32 @@ export const useProofRequest = () => {
   const canSubmit = isSuccess && anonSet !== null && proofRequest !== null
 
   useEffect(() => {
-    if (message === '' || rawSignature === undefined || anonSet === null) return
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    ;(async () => {
+      if (
+        message === '' ||
+        typeof rawSignature === 'undefined' ||
+        anonSet === null ||
+        typeof address === 'undefined'
+      )
+        return
 
-    setProofRequest(
-      new ProofRequest({
-        addresses: anonSet,
-        message,
-        rawSignature,
-        url: config.urls.proveApi,
-      }),
-    )
+      const merkleProofBytes = await generateMerkleProof(
+        anonSet,
+        address.toLowerCase(),
+        15,
+      )
+
+      setProofRequest(
+        new ProofRequest({
+          addresses: anonSet,
+          merkleProof: merkleProofBytes,
+          message,
+          rawSignature,
+          url: config.urls.proveApi,
```

Comment:
The same as above, that would be resolved in #302 
Also that shouldn't affect the new proving way since in `useProofResult.ts` I am just fetching data from `proofRequest` But I dont do any calls from it like submit. Instead I call `const { proveMembership } = useSpartanEcdsaWorker()`  which is also in the client.